packages/vats/src/vat-bank.js

```diff
@@ -171,6 +171,138 @@ export function buildRootObject(_vatPowers) {
+            /** @type {bigint} */
+            let lastBalanceUpdate = -1n;
+            /** @type {BalanceUpdater} */
+            const balanceUpdater = (value, nonce = undefined) => {
+              if (nonce !== undefined) {
+                const thisBalanceUpdate = BigInt(nonce);
+                if (thisBalanceUpdate <= lastBalanceUpdate) {
+                  return;
+                }
+                lastBalanceUpdate = thisBalanceUpdate;
+              }
+              // Convert the string value to a bigint.
+              const amt = amountMath.make(brand, BigInt(value));
+              updater.updateState(amt);
+            };
+
+            // Get the initial balance.
+            addressToUpdater.init(address, balanceUpdater);
+            const balanceString = await bankCall({
+              type: 'VPURSE_GET_BALANCE',
+              address,
+              denom: assetRecord.denom,
+            });
+            balanceUpdater(balanceString);
+
+            // Create and return the virtual purse.
+            const vpc = makePurseController(
+              bankCall,
+              assetRecord.denom,
+              brand,
+              address,
+              notifier,
+              updateBalances,
+            );
+            const vpurse = makeVirtualPurse(vpc, assetRecord);
+            brandToVPurse.init(brand, vpurse);
+            return vpurse;
+          },
+        });
+        addressToBank.init(address, bank);
+        accountsUpdater.updateState([...addressToBank.keys()]);
+        return bank;
+      };
+
+      const bankDepositFacet = Far('bankDepositFacet', {
+        getAccountsNotifier() {
+          return accountsNotifier;
+        },
+        /**
+         * Send many independent deposits.  If any of them fail, then you should
+         * reclaim the corresponding payments since they didn't get deposited.
+         *
+         * @param {Array<string>} accounts
+         * @param {Array<Payment>} payments
+         * @returns {Promise<PromiseSettledResult<Amount>[]>}
+         */
+        depositMultiple(accounts, payments) {
+          /**
+           * @param {string} account
+           * @param {Payment} payment
+           */
+          const doDeposit = async (account, payment) => {
+            // We can get the alleged brand, because the purse we send it to
+            // will do the proper verification as part of deposit.
+            const brand = await E(payment).getAllegedBrand();
+            const bank = getBankForAddress(account);
+            const purse = bank.getPurse(brand);
+            return E(purse).deposit(payment);
+          };
+
+          // We want just a regular iterable that yields deposit promises.
+          function* generateDepositPromises() {
+            const max = Math.max(accounts.length, payments.length);
+            for (let i = 0; i < max; i += 1) {
+              // Create a deposit promise.
+              yield doDeposit(accounts[i], payments[i]);
```

Comment:
Dean made `doDeposit` not be async. Does this still need to be a generator in that case?  I tried to comment in Dean's PR, but this looks like unchanged code there.

Reply:
I still want it to be async (even if there are no `await`s within it), so that a failure doesn't abort the turn.

@dtribble is that okay?